### PR TITLE
Format negative datetimes with rfc3339 (#9501)

### LINKED ISSUE #9501: displaying BC datetime causes panic
### Describe the bug

A BC datetime can be calculated and stored in a variable, but trying to display it kills nushell with a panic.

### How to reproduce

```shell
$ nu

> 0000-01-02
Sun, 02 Jan 0000 00:00:00 +0000 (2024 years ago)

> 0000-01-02 - 1day
Sat, 01 Jan 0000 00:00:00 +0000 (2024 years ago)

> 0000-01-02 - 2day
thread 'main' panicked at 'writing rfc2822 datetime to string should never fail: Error', /home/home/.cargo/registry/src/index.crates.io-6f17d22bba15001f/chrono-0.4.26/src/datetime/mod.rs:668:14
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

$
```

```shell
$ nu

> let dt = 0000-01-01 - 1day

> $dt
thread 'main' panicked at 'writing rfc2822 datetime to string should never fail: Error', /home/home/.cargo/registry/src/index.crates.io-6f17d22bba15001f/chrono-0.4.26/src/datetime/mod.rs:668:14
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

$
```

### Expected behavior

I don't mind if BC datetimes can't be formatted or even if they can't be created at all, but they ideally shouldn't panic the shell.

### Screenshots

_No response_

### Configuration

| key                | value                                                      |
| ------------------ | ---------------------------------------------------------- |
| version            | 0.81.0                                                     |
| branch             |                                                            |
| commit_hash        |                                                            |
| build_os           | linux-x86_64                                               |
| build_target       | x86_64-unknown-linux-gnu                                   |
| rust_version       | rustc 1.70.0 (90c541806 2023-05-31) (Fedora 1.70.0-1.fc38) |
| cargo_version      | cargo 1.70.0                                               |
| build_time         | 2023-06-19 15:57:54 -06:00                                 |
| build_rust_channel | release                                                    |
| features           | default, sqlite, trash, which, zip                         |
| installed_plugins  |                                                            |

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Taywee**
- fixes #9501 

# Description

`chrono::Datetime::to_rfc2822()` panics when it would format a negative year.  3339 does not.  This makes negative-year datetimes inconsistent compared to positive ones, but it's better than a panic.

### CODE REVIEW

File: crates/nu-protocol/src/value/mod.rs [hunk 1]
```diff
@@ -12,7 +12,7 @@ use crate::engine::EngineState;
 use crate::ShellError;
 use crate::{did_you_mean, BlockId, Config, Span, Spanned, Type, VarId};
 use byte_unit::ByteUnit;
-use chrono::{DateTime, Duration, FixedOffset};
+use chrono::{DateTime, Datelike, Duration, FixedOffset};
```
Comment (amtoine) [inline crates/nu-protocol/src/value/mod.rs:15]: do we need `chrono::DateLike` to call `.to_rfc3339`?

**Review by amtoine — CHANGES_REQUESTED** [review]
i was able to check this solves the issue :ok_hand: 

i've just got a question and also **could you add a test** that makes sure this issue is solved? :yum:

File: crates/nu-protocol/src/value/mod.rs (hunk 1 above)
Comment (Taywee) [reply crates/nu-protocol/src/value/mod.rs:15]: No, but we do need it to call `.year()`.

**Comment by Taywee** [comment]
I'd be happy to add a test.  I'll do that right now.

File: crates/nu-protocol/src/value/mod.rs (hunk 1 above)
Comment (amtoine) [reply crates/nu-protocol/src/value/mod.rs:15]: ooooh i see, thanks for the explaination :+1:

**Comment by Taywee** [comment]
I have added tests and validated that the negative year test fails in the previous state and succeeds with the patch.

I had to cut off the humanized stamp, because it's relative to the current time and therefore not test stable, but I don't feel too bad about it, for a small hack.

**Review by fdncred — APPROVED** [review]
Thanks! I'm good with these changes.

**Review by amtoine — APPROVED** [review]
thanks @Taywee, only a little merge resolution and that should be ok :yum:

**Comment by Taywee** [comment]
Merge conflict resolved.  I rebased it for a cleaner history.

**Comment by amtoine** [comment]
@Taywee 
no need to rebase inside a PR :wink: 
- it breaks the PR review process because the commits that have been reviewed are not part of the PR anymore
- we squash and rebase the PR onto the `main` branch anyway

**Comment by Taywee** [comment]
My mistake.  To be completely honest, I also tried to fix the merge conflict in the GitHub editor and managed to break the PR in the process, so I also rebased to remove the ugly broken commit from the history, but if nu PRs are always rebased and squashed anyway, I suppose that didn't matter. :sweat_smile: 

I'll avoid unnecessary rebases in the future.  Thanks for the heads up.

**Comment by fdncred** [comment]
Thanks!

**Pull request merged**